From e5a808baab27f050b369b3f6bd395e5972e5236f Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:35 -0400
Subject: [block] blk-flush: clear flush_rq's tag in flush_end_io()

Message-id: <1402670238-13916-108-git-send-email-snitzer@redhat.com>
Patchwork-id: 83837
O-Subject: [RHEL7.1 PATCH 107/150] block: mq flush: clear flush_rq's tag in flush_end_io()
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 14b83e172f0bc83b8dcf78ee8b1844beeffb418d
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Thu Jun 5 00:23:55 2014 +0800

    block: mq flush: clear flush_rq's tag in flush_end_io()

    blk_mq_tag_to_rq() needs to be able to tell if it should return
    the original request, or the flush request if we are doing a flush
    sequence. Clear the flush tag when IO completes for a flush, since
    that is what we are comparing against.

    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-flush.c b/block/blk-flush.c
index adf85a2..8cb6210 100644
--- a/block/blk-flush.c
+++ b/block/blk-flush.c
@@ -225,7 +225,7 @@ static void flush_end_io(struct request *flush_rq, int error)
 
  if (q->mq_ops) {
   spin_lock_irqsave(&q->mq_flush_lock, flags);
-  q->flush_rq->cmd_flags = 0;
+  q->flush_rq->tag = -1;
  }
 
  running = &q->flush_queue[q->flush_running_idx];
-- 
1.7.1